### PR TITLE
Bug 1162138 - [accessibility][gaia-button] Make sure gaia-button is accessible to screen reader

### PR DESCRIPTION
**Pull request by gasolin**

### CODE REVIEW

File: gaia-button.js [hunk 1]
```diff
@@ -16,8 +16,30 @@ module.exports = component.register('gaia-button', {
...
+  },
+
+  makeAccessible: function() {
     this.setAttribute('role', 'button');
+
+    // Make tabable
     this.tabIndex = 0;
+
+    // let screen reader read the button is disabled
+    if (this.hasAttribute('disabled')) {
```
Comment (yzen) [inline gaia-button.js:32]: :+1:

File: gaia-button.js [hunk 2]
```diff
@@ -16,8 +16,30 @@ module.exports = component.register('gaia-button', {
...
+      this.setAttribute('aria-disabled', true);
+    }
+    // let screen reader read the button text and ignore the icon text
```
Comment (wilsonpage) [inline gaia-button.js:35]: NIT: Add space above

File: gaia-button.js [hunk 3]
```diff
@@ -16,8 +16,30 @@ module.exports = component.register('gaia-button', {
...
+    // when both icon and text existed
+    if (this.childElementCount > 1 && this.getElementsByTagName('i')) {
+      var items = this.getElementsByTagName('i');
```
Comment (wilsonpage) [inline gaia-button.js:38]: `this.getElementsByTagName('i')` called twice.

Cleaner alternative:

``` js
var icons = this.querySelectorAll('i');
[].forEach.call(icons, icon => icon.setAttribute('aria-hidden', true);
```

File: gaia-button.js [hunk 4]
```diff
@@ -16,8 +16,30 @@ module.exports = component.register('gaia-button', {
 
   created: function() {
     this.setupShadowRoot();
+    // offload the initiate operation to save loadtime
+    setTimeout(() => {
+      this.makeAccessible();
+    });
```
Comment (wilsonpage) [inline gaia-button.js:22]: Explain _why_ this is faster

``` js
setTimeout(() => this.makeAccessible());
```

**Comment by gasolin** [comment]
will replace by new proposal

**Pull request closed without merging**